src/basic_recipes/spy.jl

```diff
@@ -12,24 +15,37 @@ spy(0..1, 0..1, x)
 ```
 """
 @recipe Spy (x, y, z) begin
-    marker = automatic
+    marker = Rect
     markersize = automatic
     framecolor = :black
+    framevisible = true
     framesize = 1
+    color = nothing
     MakieCore.mixin_generic_plot_attributes()...
     MakieCore.mixin_colormap_attributes()...
 end
 
-function convert_arguments(::Type{<: Spy}, x::SparseArrays.AbstractSparseArray)
-    (0..size(x, 1), 0..size(x, 2), x)
+function data_limits(plot::Spy)
+    data_limits(plot.plots[2])
 end
-function convert_arguments(::Type{<: Spy}, x, y, z::SparseArrays.AbstractSparseArray)
-    (x, y, z)
+
+function boundingbox(p::Spy, space::Symbol=:data)
+    boundingbox(p.plots[2], space)
+end
+
+function convert_arguments(::Type{<:Spy}, matrix::AbstractMatrix{T}) where T
+    Tr = Makie.float_type(T)
+    return convert_arguments(Spy, Tr.((0, size(matrix, 1))), Tr.((0, size(matrix, 2))), matrix)
 end
 
-function calculated_attributes!(::Type{<: Spy}, plot)
+function convert_arguments(::Type{<:Spy}, xs, ys, matrix::AbstractMatrix)
+    x = to_interval(xs, "x")
+    y = to_interval(ys, "y")
+    return (x, y, convert(SparseArrays.SparseMatrixCSC, matrix))
 end
 
+needs_tight_limits(::Spy) = true
+
 function plot!(p::Spy)
     rect = lift(p, p.x, p.y) do x, y
         xe = extrema(x)
```

Comment:
Maybe better to use `minmax(endpoints(x)...)` to allow reversed Intervals since `extrema(1..0)` errors?

Reply:
Hm yeah, but reversing still won't work since we won't actually  look up the points in reverse.

Reply:
Oh right, this should just be left/rightendpoint. I'm not sure if rect allows negative widths, so maybe we should replace it with a (x, y, w, h) tuple?